### PR TITLE
Feature/prep db for multi person

### PR DESCRIPTION
**Pull request by joecox**
## This PR...
- Preps the DB/controllers/client for multi person Kudos
- Migrates the DB to use a kudo-receiver join table
- Splits exiting multi-recipients into multiple join rows
- Shows only the first recipient in UI
- Writes only first email submitted as receiver

## How Has This Been Tested?
👀

## Types of changes

- [x] Refactor (non-breaking update to code that doesn't change functionality or external behavior, may include some new code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: db/migrate/20180507172336_multi_person_kudos.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+          User.where(email: emails).map do |user|
+            { kudo_id: kudo.id, receiver_id: user.id }
+          end
+        end.flatten
+
+        KudoReceipt.create!(receipt_rows)
+
+        remove_column :kudos, :receiver_id
+        User.where("email LIKE '%,%'").destroy_all
```
Comment (trisarahsorus) [inline db/migrate/20180507172336_multi_person_kudos.rb]: What is this line doing?

File: db/schema.rb [hunk 2]
```diff
@@ -20,11 +20,18 @@
...
+    t.uuid     "giver_id",   null: false
+  end
+
+  create_table "kudos_to_receivers", id: :uuid, default: -> { "uuid_generate_v4()" }, force: :cascade do |t|
     t.datetime "created_at",  null: false
     t.datetime "updated_at",  null: false
-    t.uuid     "giver_id",    null: false
+    t.uuid     "kudo_id",     null: false
     t.uuid     "receiver_id", null: false
+    t.index ["kudo_id", "receiver_id"], name: "index_kudos_to_receivers_on_kudo_id_and_receiver_id", unique: true, using: :btree
```
Comment (trisarahsorus) [inline db/schema.rb]: Should we add another index on `receiver_id`?

File: db/schema.rb [hunk 3]
```diff
@@ -20,11 +20,18 @@
   enable_extension "uuid-ossp"
 
   create_table "kudos", id: :uuid, default: -> { "uuid_generate_v4()" }, force: :cascade do |t|
-    t.text     "body",        null: false
+    t.text     "body",       null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.uuid     "giver_id",   null: false
```
Comment (trisarahsorus) [inline db/schema.rb]: As Kudos usage increases, I think we'll want indexes on `giver_id` and `created_at`.
(Maybe not in this PR, but eventually!)

File: app/controllers/kudos_controller.rb [hunk 4]
```diff
@@ -60,7 +60,7 @@ def create
 
     kudo = Kudo.new(
       giver_id: giver_id,
-      receiver_id: receiver.id,
+      receiver_ids: [receiver.id],
```
Comment (trisarahsorus) [inline app/controllers/kudos_controller.rb]: Confused, should this be `receivers`?

Comment (nnsookwon) [reply app/controllers/kudos_controller.rb]: I believe `receivers` refers to the actual `User` models, whereas this is just using the Users' ids. I think passing in an array for `receiver_ids` auto magically creates the multiple `KudoReceipts`s to join between `Kudo` and `User` (@joecox please confirm)

File: db/schema.rb (hunk 2 above)
Comment (joecox) [reply db/schema.rb]: Yeah, good idea!

File: db/schema.rb (hunk 3 above)
Comment (joecox) [reply db/schema.rb]: I agree, the `created_at` index should be especially useful with paginated scrolling later.

File: app/controllers/kudos_controller.rb (hunk 4 above)
Comment (trisarahsorus) [reply app/controllers/kudos_controller.rb]: Gotcha, thanks for the explanations @joecox and @nnsookwon!
Rails magic continues to befuddle me. =_=

The logical separation sounds good, thanks for the smaller PRs for us! :D

**Pull request merged**